content/3_Silver/Prefix_Sums_2.mdx

```diff
@@ -439,6 +439,108 @@ int main() {
 
 </CPPSection>
 
+<JavaSection>
+
+```java
+import java.io.DataInputStream;
+import java.io.IOException;
+import java.io.PrintWriter;
+ 
+public class ForestQueries3 {
+    public static void main(String[] args) throws IOException {
+        Reader r = new Reader();
+        PrintWriter pw = new PrintWriter(System.out);
+ 
+        int n = r.nextInt();
+        int q = r.nextInt();
+        int N = n + 1;
+        int[][] forest = new int[N][N];
```

Comment:
Is this well defined behavior (I found out today it's not in cpp, but IDK about java)

Reply:
Just wandered across this thread, and the answer is [yes, it's well defined in Java](https://stackoverflow.com/a/3426854/10232025) :)

Reply:
ok, thanks